versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/CommitImpl.java

```diff
@@ -186,11 +199,29 @@ CommitResult<Commit> commit(
 
     fromCommitMeta(metadata, commit);
 
+    CommitObj newHead;
     try {
-      CommitObj newHead = commitLogic.doCommit(commit.build(), objectsToStore);
-
+      CreateCommit createCommit = commit.build();
+      newHead =
+          commitLogic.buildCommitObj(
+              createCommit,
+              c -> CONFLICT,
+              (k, v) -> {},
+              NO_VALUE_REPLACEMENT,
+              NO_VALUE_REPLACEMENT);
+
+      // If 'commitRetryState.storedContents' already contains the commit-ID, __we__ already
+      // successfully persisted that commit. This can happen, if the `Persist` implementation raised
+      // an `UnknownOperationResultException` in one of the following operations (reference bump for
+      // example).
+      boolean stored = commitLogic.storeCommit(newHead, objectsToStore);
+      if (stored) {
+        commitRetryState.commitPersisted = newHead.id();
+        commitRetryState.storedContents.addAll(toStore);
```

Comment:
Is `commitRetryState.storedContents` still needed? I could not find where its value is read :thinking:

Reply:
Like 141 is a comment in this file :thinking:

Reply:
There's a `if (storedContents.add(...))`

Reply:
Sorry, I really cannot find that `if/add` :sweat_smile:

Reply:
I meant line 181 - but that's somehow not exactly how I always thought it is